### PR TITLE
Honda: move to platformconfig

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: selfdrive/car/honda/carcontroller.py [hunk 1]
```diff
@@ -164,12 +164,12 @@ def update(self, CC, CS, now_nanos):
     can_sends = []
 
     # tester present - w/ no response (keeps radar disabled)
-    if self.CP.carFingerprint in (HONDA_BOSCH - HONDA_BOSCH_RADARLESS) and self.CP.openpilotLongitudinalControl:
+    if (self.CP.flags & HondaFlags.BOSCH) and (not self.CP.flags & HondaFlags.BOSCH_RADARLESS) and self.CP.openpilotLongitudinalControl:
```
Comment (sshane) [inline selfdrive/car/honda/carcontroller.py:167]:
```suggestion
    if (self.CP.flags & (HondaFlags.BOSCH | HondaFlags.BOSCH_RADARLESS) == HondaFlags.BOSCH) and self.CP.openpilotLongitudinalControl:
```

works too, not sure if cleaner

File: selfdrive/car/honda/interface.py [hunk 2]
```diff
@@ -33,12 +32,12 @@ def get_pid_accel_limits(CP, current_speed, cruise_speed):
...
 
   @staticmethod
-  def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
+  def _get_params(ret, candidate: CAR, fingerprint, car_fw, experimental_long, docs):
     ret.carName = "honda"
 
     CAN = CanBus(ret, fingerprint)
 
-    if candidate in HONDA_BOSCH:
+    if candidate.config.flags & HondaFlags.BOSCH:
```
Comment (sshane) [inline selfdrive/car/honda/interface.py:40]: You can create/set these flags in a new PR to reduce the diff if you'd like

File: selfdrive/car/honda/values.py [hunk 3]
```diff
@@ -49,7 +49,16 @@ class HondaFlags(IntFlag):
   # Bosch models with alternate set of LKAS_HUD messages
```
Comment (sshane) [inline selfdrive/car/honda/values.py:49]:
```suggestion
  # Detected flags
  # Bosch models with alternate set of LKAS_HUD messages
```

File: selfdrive/car/honda/values.py [hunk 4]
```diff
@@ -49,7 +49,16 @@ class HondaFlags(IntFlag):
...
   BOSCH_EXT_HUD = 1
   BOSCH_ALT_BRAKE = 2
+  BOSCH = 4
```
Comment (sshane) [inline selfdrive/car/honda/values.py:52]:
```suggestion

  # Static flags
  BOSCH = 4
```

File: selfdrive/car/honda/interface.py [hunk 5]
```diff
@@ -278,27 +277,27 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
...
 
-    if candidate in HONDA_BOSCH_RADARLESS:
+    if candidate.config.flags & HondaFlags.BOSCH_RADARLESS:
       ret.safetyConfigs[0].safetyParam |= Panda.FLAG_HONDA_RADARLESS
 
     # min speed to enable ACC. if car can do stop and go, then set enabling speed
     # to a negative value, so it won't matter. Otherwise, add 0.5 mph margin to not
     # conflict with PCM acc
-    ret.autoResumeSng = candidate in (HONDA_BOSCH | {CAR.CIVIC}) or ret.enableGasInterceptor
+    ret.autoResumeSng = bool(candidate.config.flags & HondaFlags.AUTORESUME_SNG) or ret.enableGasInterceptor
```
Comment (sshane) [inline selfdrive/car/honda/interface.py:300]: Perhaps for an exception of one we don't need a whole flag?

File: selfdrive/car/honda/carstate.py [hunk 6]
```diff
@@ -39,20 +37,17 @@ def get_can_messages(CP, gearbox_msg):
...
-    messages.append((gearbox_msg, 50))
-  else:
-    messages.append((gearbox_msg, 100))
+  messages.append((gearbox_msg, 50))
 
   if CP.flags & HondaFlags.BOSCH_ALT_BRAKE:
     messages.append(("BRAKE_MODULE", 50))
 
-  if CP.carFingerprint in (HONDA_BOSCH | {CAR.CIVIC, CAR.ODYSSEY, CAR.ODYSSEY_CHN}):
+  if CP.flags & (HondaFlags.BOSCH | HondaFlags.ELECTRIC_PARKING_BRAKE):
```
Comment (sshane) [inline selfdrive/car/honda/carstate.py:45]: Did you forget to add the flag to the Odysseys?

File: selfdrive/car/honda/values.py [hunk 7]
```diff
@@ -49,7 +49,16 @@ class HondaFlags(IntFlag):
...
+  BOSCH_RADARLESS = 8
 
+  NIDEC = 16
+  NIDEC_ALT_PCM_ACCEL = 32
+  NIDEC_ALT_SCM_MESSAGES = 64
+  NIDEC_RADARLESS = 128
```
Comment (sshane) [inline selfdrive/car/honda/values.py:58]: What's Nidec radarless?

File: selfdrive/car/honda/interface.py (hunk 5 above)
Comment (jnewb1) [reply selfdrive/car/honda/interface.py:300]: Hmm, I guess but I think ideally carcontoller doesn't even know what a civic is

File: selfdrive/car/honda/carcontroller.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/car/honda/carcontroller.py:167]: I think that's where the 'self.flags' is useful. I think having that everywhere is way too verbose

File: selfdrive/car/honda/values.py (hunk 7 above)
Comment (jnewb1) [reply selfdrive/car/honda/values.py]: removed

File: selfdrive/car/honda/values.py (hunk 4 above)
Comment (jnewb1) [reply selfdrive/car/honda/values.py:55]: done

File: selfdrive/car/honda/values.py (hunk 3 above)
Comment (jnewb1) [reply selfdrive/car/honda/values.py:50]: done

File: selfdrive/car/honda/interface.py (hunk 2 above)
Comment (jnewb1) [reply selfdrive/car/honda/interface.py]: yea, done

File: selfdrive/car/honda/interface.py (hunk 5 above)
Comment (jnewb1) [reply selfdrive/car/honda/interface.py]: like ideally the platform config tells carcontroller and carstate everything it needs to know to control the car

**Comment by jnewb1** [comment]
diff is the flags:

image

**Pull request merged**